### PR TITLE
Saved signup data to mongodb + compare login input to mongodb data

### PR DESCRIPTION
**Pull request by ErmiasMerine**

### CODE REVIEW

File: back-end/app.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ let bcrypt = require("bcrypt");
 let express = require("express");
 let fs = require("fs/promises");
 let morgan = require("morgan");
-
+const cors = require('cors');
```
Comment (huang-a) [inline back-end/app.js:6]: No changes needed: I think the cors import could be removed and have another header be added after L30, but this is fine as-is.

File: back-end/app.js [hunk 2]
```diff
@@ -95,7 +96,7 @@ server.post("/signup", (req, resp) => {
 server.post("/login", (req, resp) => {
 	let username = req.body.username ?? "";
 	let password = req.body.password ?? "";
-
+	console.log("login body", req.body)
```
Comment (huang-a) [inline back-end/app.js:99]: Probably shouldn't send user credentials to stdout, this exposes passwords if logs are saved, etc.

File: front-end/src/screen/signIn.js [hunk 3]
```diff
@@ -1,27 +1,61 @@
...
+    const [password, setpassword] = useState("initialState")
+
+    const handleSubmit = (event) => {
+      event.preventDefault(); 
+      const payload = {
+          username: username,
+          password: password
+        };
+        axios({
+          url: 'http://localhost:61001/login',
```
Comment (huang-a) [inline front-end/src/screen/signIn.js:15]: L15: We should use the backend base url from the config to make sure this works in deployment as well.

File: front-end/src/screen/signUp.js [hunk 4]
```diff
@@ -1,34 +1,82 @@
...
+    const handleSubmit = (event) => {
+      event.preventDefault(); 
+      const payload = {
+          first_name: fname,
+          last_name: lName,
+          username: username,
+          password: password
+        };
+      axios({
+        url: 'http://localhost:61001/signup',
```
Comment (huang-a) [inline front-end/src/screen/signUp.js:19]: L19: We should use the backend base url from the config to make sure this works in deployment as well.

**Review by huang-a — CHANGES_REQUESTED** [review]
Mostly looks good. Just a few changes that deal with deployment.

File: front-end/src/screen/signIn.js [hunk 5]
```diff
@@ -1,27 +1,61 @@
...
+    const [password, setpassword] = useState("initialState")
+
+    const handleSubmit = (event) => {
+      event.preventDefault(); 
+      const payload = {
+          username: username,
+          password: password
+        };
+        axios({
+          url: 'http://127.0.0.1:61001/login',
```
Comment (huang-a) [inline front-end/src/screen/signIn.js:15]: L15 + another line: should be `${config.backend_base_url}` instead of `127.0.0.1:61001`

**Review by huang-a — CHANGES_REQUESTED** [review]
Oops, missed this the first time around.

File: front-end/src/screen/signIn.js [hunk 6]
```diff
@@ -1,27 +1,61 @@
...
+        };
+        axios({
+          url: 'http://localhost:61001/login',
+          method: 'POST',
+          data: payload
+        })
+          .then(() => {
+            console.log('Data has been sent to the server');
+            this.resetUserInputs();
+            this.getBlogPost();
```
[the remark this answers is not in the document]
Comment (ErmiasMerine) [reply front-end/src/screen/signIn.js:22]: remains of a feature i wanted to implement but decided not to

[the remark this answers is not in the document]
Comment (huang-a) [reply front-end/src/screen/signIn.js]: ah gotcha, cool

**Review by huang-a — APPROVED** [review]
looks good to me

**Pull request merged**